lib/services/android-project-service.ts

```diff
@@ -118,19 +120,135 @@ class AndroidProjectService implements IPlatformProjectService {
+        var metadataDirName = "__metadata";
+        var inputDir = libsFolder;
+        var outDir = path.join(libsFolder, metadataDirName);
+        this.$fs.ensureDirectoryExists(outDir).wait();
+
+        shell.cp("-f", path.join(__dirname, "../../resources/tools/android.jar"), inputDir);
+        shell.cp("-f", path.join(__dirname, "../../resources/tools/android-support-v4.jar"), inputDir);
+        shell.cp("-f", path.join(projectRoot, "libs/*.jar"), inputDir);
+
+        this.spawn('java', ['-jar', metadataGeneratorPath, inputDir, outDir]).wait();
+    }
 
 	public buildProject(projectRoot: string): IFuture<void> {
-		return (() => {
+        return (() => {
 			var buildConfiguration = options.release ? "release" : "debug";
-			var args = this.getAntArgs(buildConfiguration, projectRoot);
-			this.spawn('ant', args).wait();
+            var args = this.getAntArgs(buildConfiguration, projectRoot);
+            var args2 = this.getAntArgs(buildConfiguration, projectRoot);
+            this.spawn('ant', args).wait();
+            this.generateMetadata(projectRoot);
+            this.updateMetadata(projectRoot);
+            // build the project again in order to include the newly generated metadata
+            this.spawn('ant', args2).wait();
 		}).future<void>()();
 	}
 
 	public isPlatformPrepared(projectRoot: string): IFuture<boolean> {
 		return this.$fs.exists(path.join(projectRoot, "assets", constants.APP_FOLDER_NAME));
-	}
+    }
+
+    private generateBuildFile(projDir: string, targetSdk: string): void {
+        this.$logger.info("Generate build.xml for %s", projDir);
+        var cmd = util.format("android update project -p %s --target %s --subprojects", projDir, targetSdk);
+        this.$childProcess.exec(cmd).wait();
+    }
+
+    private parseProjectProperties(projDir: string, destDir: string): void {
+
+        var projProp = path.join(projDir, "project.properties");
+
+        if (!this.$fs.exists(projProp).wait()) {
+            this.$logger.warn("File %s does not exist", projProp);
+            return;
+        }
+
+        var lines = fs.readFileSync(projProp, { encoding: "utf-8" }).split(os.EOL);
+        var thiz = this;
+
+        lines.forEach((elem, idx, arr) => {
+            var match = elem.match(/android\.library\.reference\.(\d+)=(.*)/);
+            if (match) {
+                var libRef: ILibRef = { idx: parseInt(match[1]), path: match[2].trim() };
+                libRef.adjustedPath = thiz.$fs.isRelativePath(libRef.path) ? path.join(projDir, libRef.path) : libRef.path;
+                thiz.parseProjectProperties(libRef.adjustedPath, destDir);
+            }
+        });
+
+        this.$logger.info("Copying %s", projDir);
+        shell.cp("-Rf", projDir, destDir);
+
+        var targetDir = path.join(destDir, path.basename(projDir));
+        // TODO: parametrize targetSdk
+        var targetSdk = "android-17";
```

Comment:
We always generate build file using `android-17`. I was wondering if we shouldn't get the latest installed android sdk and generate build file using it.